### ISSUE TITLE
Stopped receiving messages, duplicated slackts entries

### ISSUE THREAD

**Issue by antonok-edm** [matrix-org/matrix-appservice-slack#363]
I started getting the following error yesterday whenever a new message comes in:

```
Mar-17 16:14:43.182 ERROR BridgedRoom Failed to process event
Mar-17 16:14:43.184 ERROR BridgedRoom QueryResultError {
    code: queryResultErrorCode.multiple
    message: "Multiple rows were not expected."
    received: 2
    query: "SELECT * FROM events WHERE slackChannel = '<redacted>' AND slackTs = '1584288956.048300'"
}
```

Executing the query manually using `psql` gives the following results:

```
psql=> SELECT * FROM events WHERE slackChannel = '<redacted>' AND slackTs = '1584288956.048300';
             roomid                 |                   eventid                    | slackchannel |      slackts      | extras
------------------------------------+----------------------------------------------+--------------+-------------------+--------
 !r0oMId:homeserver.com             | $8QCvLNd4yWJbPBtTtl7-J7eSgKp23BFuL-H5-1GTnC8 | <redacted>   | 1584288956.048300 | {}
 !r0oMId:homeserver.com             | $3m_F6koP5-rYWSFztm9i-hg9NtNaGE2pw6k8u8dKOMQ | <redacted>   | 1584288956.048300 | {}
(2 rows)
```

This query stops new messages from being received in the bridged channel, as it's executed every time one should be received.

I've been running this bridge in the docker container (image id `37aacd1b4011`) for about 4 months without issue until yesterday. After seeing this in the logs I tried updating to the latest available image (`847bed58183e`), but it still fails the same way.

Happy to provide further information to further debug the issue, but I'd also appreciate it if there's a way I can manually patch the state of the database in the meantime.

**Comment by Half-Shot** [comment]
You can delete one of the affected events from that table to clear up the issue. I suspect the bug is that we insert multiple events into the DB if we represent a slack messsage with multiple matrix messages, but we are fetching a single event back out which breaks it.

**Comment by antonok-edm** [comment]
Makes sense. Do you have any intuition on how to figure out which event of the two would be okay to remove? Or should it be fine either way?

**Comment by Half-Shot** [comment]
It doesn't matter which one you delete, since they are only used for reactions/edit/replies bridging. If you don't interact with those events, it won't matter too much.
